src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  for (let i = 0; i < items.length; i += 1) {
+    const item = items[i];
+    const keyValue = String(item[key]);
+
+    if (!groupedItems[keyValue]) {
+      groupedItems[keyValue] = [];
+    }
+
+    groupedItems[keyValue].push(item);
+  }
```

Comment:
You can use method forEach
When I did a for loop, TS fixed me

Reply:
Good point, use `forEach` here